### DISCUSSION TITLE
Download file from s3 url

### DISCUSSION THREAD

**Discussion by luna215** [sindresorhus/ky#449] [Q&A] [answered]
Hi, how would I go about downloading a file from an s3 url?  I thought it was a simple `ky.get(...)` liner, but I keep getting this error on the 

```
Access to fetch at 'https://molstar-example.s3.us-west-1.amazonaws.com/2ym3_protein_processed.pdb' from origin 'null' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

Am I using the wrong operations/functions?

**Comment by sholladay** [comment, marked as answer]
As with most services, S3 does not respond with CORS headers by default, because doing so inherently weakens the security model of the browser. You have to explicitly enable CORS on the S3 side for cross origin requests to work.

See:
 - https://docs.aws.amazon.com/AmazonS3/latest/userguide/cors.html
 - https://stackoverflow.com/questions/17533888/s3-access-control-allow-origin-header
 - https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
